service/internal/asg/asg.go

```diff
@@ -181,6 +171,10 @@ func (a *ASG) lookupASGs(ctx context.Context, names []string) ([]*autoscaling.Gr
 		return nil, microerror.Mask(err)
 	}
 
+	if len(names) == 0 {
```

Comment:
This is a minor optimization. In case there will ever be no name provided we do not even have to ask the AWS API.